src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,145 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+import { Person } from '../../types';
+
+interface Props {
+  people: Person[];
+  selectedSlug: string;
+}
+
+const nameSpacesToDashes = (spacedName: string) => {
+  return spacedName.toLowerCase().split(' ').join('-');
+};
+
+const findByName = (peopleArray: Person[], nameFindBy: string | '') => {
+  return peopleArray.find(person => person.name === nameFindBy) || null;
+};
+
+export const PeopleTable: FC<Props> = ({ people, selectedSlug }) => {
+  const existingNames = people.map(person => person.name);
+
+  return (
+    <table
+      data-cy="peopleTable"
+      className="
+        table
+        is-striped
+        is-hoverable
+        is-narrow
+        is-fullwidth
+      "
+    >
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Sex</th>
+          <th>Born</th>
+          <th>Died</th>
+          <th>Mother</th>
+          <th>Father</th>
+        </tr>
+      </thead>
+
+      <tbody>
+
+        {people.map(({
+          name,
+          sex,
+          born,
+          died,
+          motherName,
+          fatherName,
+        }) => {
+          const slug = `${nameSpacesToDashes(name)}-${born}`;
+
+          const isSelected = slug === selectedSlug;
+
+          const motherNotEmpty = motherName !== '' && motherName !== null;
+          const motherInTable = motherNotEmpty
+            ? existingNames.includes(motherName || 'not-a-name-but-string')
+            : false;
+          const mother = motherInTable
+            ? findByName(people, motherName || 'not-a-name-but-string')
+            : null;
+          const motherSlug = motherInTable
+            ? `${nameSpacesToDashes(motherName || 'not-a-name-but-string')}-${mother?.born}`
+            : null;
+
+          const fatherNotEmpty = fatherName !== '' && fatherName !== null;
+          const fatherInTable = fatherNotEmpty
+            ? existingNames.includes(fatherName || 'not-a-name-but-string')
+            : false;
+          const father = fatherInTable
+            ? findByName(people, fatherName || 'not-a-name-but-string')
+            : null;
+          const fatherSlug = fatherInTable
+            ? `${nameSpacesToDashes(fatherName || 'not-a-name-but-string')}-${father?.born}`
+            : null;
```

Comment:
Looks like `костиль`, instead of iterating arrays to get names and then searching for names here, prepare list of people before mapping, find `mother` and `father` in list and add these objects as Person properties (if no objects was found - set null)
Then handle this data to render needed link or string